bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -277,10 +284,11 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 				return r.rotateMachinePoolBootstrapToken(ctx, config, cluster, scope)
 			}
 		}
-		// In any other case just return as the config is already generated and need not be generated again.
-		return ctrl.Result{}, nil
+		// If config.Status.Ready, just return as the config is already generated and need not be generated again.
+		if config.Status.Ready {
+			return ctrl.Result{}, nil
+		}
```

Comment:
should this be 
```suggestion
		config.Status.Ready := true
		return ctrl.Result{}, nil
```

Reply:
I see config.status.Ready is set at 2 places after ensuring config.Status.DataSecretName is set . Should I remove the switch case above L260-L264 and ensure  to set here once secret already exist? otherwise it seem to be duplicated ? Thoughts?